src/CacheManager.StackExchange.Redis/RedisCacheHandle.cs

```diff
@@ -270,6 +270,13 @@ public override bool Exists(string key, string region)
         }
 
         /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keyPattern = GetKey(pattern, region);
+            return _connection.Servers.First().Keys(_redisConfiguration.Database, keyPattern).Select(k => k.ToString());
```

Comment:
see my longer comment in the issue, having a hashset approach as lookup for all keys is not something I'd like to have